pkg/netceptor/netceptor.go

```diff
@@ -730,6 +734,25 @@ func (s *Netceptor) GetServerTLSConfig(name string) (*tls.Config, error) {
 	return sc.Clone(), nil
 }
 
+// AddWorkCommand records a work command so it can be included in service announcements
+func (s *Netceptor) AddWorkCommand(command string) error {
+	if command == "" {
+		return fmt.Errorf("must provide a name")
+	}
+	s.serviceAdsLock.RLock()
+	defer s.serviceAdsLock.RUnlock()
+	if n, ok := s.serviceAdsReceived[s.NodeID()]; ok {
+		// if it's the local node, just update the local service advertisement
```

Comment:
@fosterseth this should address the problem you noticed; if the netceptor in question is on the *local* node, I just modify the local advertisement explicitly.

Otherwise for *remote* nodes, I store the record on the netceptor object so that every subsequent remote advertisement that comes in gets updated.

Reply:
For posterity:

![image](https://user-images.githubusercontent.com/214912/107293444-9a56ec00-6a39-11eb-9502-a6822b8a01b8.png)